oanda_v20_platform/oanda/oanda.py

```diff
@@ -0,0 +1,399 @@
+            return data
+        except: 
+            self.logger.exception(f"OANDA DATA ERROR - Order.get_pending_orders failed to return any pending orders")
+
+
+    def buy_market(self, units, instrument):
+        try:
+            url = self.base_url + '/v3/accounts/' + self.account + '/orders'
+            data = {
+                "order": {
+                    "units": units,
+                    "instrument": instrument,
+                    "timeInForce": "FOK",
+                    "type": "MARKET",
+                    "positionFill": "DEFAULT"
+                }
+            }
+            r = requests.post(url, headers=self.headers, json=data)
+            self.notify_order(r.json())
+        except:
+            self.logger.exception(f"OANDA DATA ERROR - Order.buy_market failed to send the order")
+
+
+    def sell_market(self, units, instrument):
+        try:
+            url = self.base_url + '/v3/accounts/' + self.account + '/orders'
+            data = {
+                "order": {
+                    "units": -units,
+                    "instrument": instrument,
+                    "timeInForce": "FOK",
+                    "type": "MARKET",
+                    "positionFill": "DEFAULT"
+                }
+            }
+            r = requests.post(url, headers=self.headers, json=data)
+            self.notify_order(r.json())
+        except:
+            self.logger.exception(f"OANDA DATA ERROR - Order.sell_market failed to send the order")
+            
+    def notify_order(self, order):
+        self.order = order
+        if 'orderCancelTransaction' in self.order:
+            print('Order Transaction Canceled:')
+            msg = f"{order['orderCancelTransaction']['type']}, {order['orderCancelTransaction']['reason']}"
+            print(msg)
+            if self.text_notifications:
+                TwilioSMS(self.twilio_sid, self.twilio_token, self.twilio_number, 
+                            self.recipient_number).send_text(msg)
+
+            self.logger.exception(f"OANDA ORDER ERROR - {msg}")
+
+        print('\n')
+        if 'orderFillTransaction' in self.order:
+            time = order["orderFillTransaction"]["time"]
+            orderID = order["orderFillTransaction"]["orderID"]
+            instrument = order["orderFillTransaction"]["instrument"]
+            units = order["orderFillTransaction"]["units"]
+            price = order["orderFillTransaction"]["price"]
+            reason = order["orderFillTransaction"]["reason"]
+            pl = order["orderFillTransaction"]["pl"]
+            msg = f'*** ORDER FULFILLED ***\nTime: {time}\nType: {reason}\nOrder Id: {orderID}\nInstrument: {instrument}\nUnits: {units}\nPrice: {price}\nP/L: ${pl}'
+            print(msg)
+            if self.text_notifications:
+                TwilioSMS(self.twilio_sid, self.twilio_token, self.twilio_number, self.recipient_number).send_text(msg)
+            self.logger.warning(f"OANDA ORDER SUCCESSFUL - {msg}")
+
```

Comment:
Any particular reason you chose to throw a log level of warning here rather than info?

Reply:
In the logging setup you can set the level that is sent to the console. I was thinking that when stuff is running smoothly it could be raised to warning so that only trades etc come through to the console, whilst the log.txt would still have the extra info for debugging etc.

Reply:
Gotcha. Yea, that logic makes sense to me. No need to clutter up debug logs with order executions, esp since oanda has that data, and I sync all my accounts up to MyFxBook anyway and they have nice charts and excel exports.